providers/unleash/pkg/provider.go

```diff
@@ -0,0 +1,168 @@
+package unleash
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Unleash/unleash-client-go/v3"
+	"github.com/Unleash/unleash-client-go/v3/api"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+const providerNotReady = "Provider not ready"
+const generalError = "general error"
+
+type Provider struct {
+	status      of.State
+	eventStream chan of.Event
```

Comment:
Given that there's no event stream usage, this can be safely removed (including `EventChannel` implementation)

From OF point of view, it is not essential for a provider to support events :)

Reply:
Thanks, removed it.
Unleash [supports events listener](https://github.com/Unleash/unleash-client-go/blob/2b58cc4de44e179461a22a29660f049d24d50d4c/unleash_test.go#L18), if you want to take it for a separate PR, by something like converting the unleash events to OpenFeature events, we can discuss separately.